src/plugins/session-key/TokenSessionKeyPlugin.sol

```diff
@@ -0,0 +1,160 @@
+/// It implements the logic for session keys that are allowed to call ERC20
+/// transferFrom function. It allows for session key owners to access MSCA
+/// with `transferFromSessionKey` function, which calls `executeFromPluginExternal`
+/// function in PluginExecutor contract.
+/// The target ERC20 contract and the selector for transferFrom function are hardcoded
+/// in this plugin, since the pluginManifest function requires the information of
+/// permitted external calls not to be changed in the future. For other child session
+/// key plugins, there can be a set of permitted external calls according to the
+/// specific needs.
+
+contract TokenSessionKeyPlugin is BasePlugin, ITokenSessionKeyPlugin {
+    string public constant NAME = "Token Session Key Plugin";
+    string public constant VERSION = "1.0.0";
+    string public constant AUTHOR = "Decipher ERC-6900 Team";
+
+    // Mock address of target ERC20 contract
+    address public constant TARGET_ERC20_CONTRACT = 0xdeaDDeADDEaDdeaDdEAddEADDEAdDeadDEADDEaD;
+    bytes4 public constant TRANSFERFROM_SELECTOR =
+        bytes4(keccak256(bytes("transferFrom(address,address,uint256)")));
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Execution functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc ITokenSessionKeyPlugin
+    function transferFromSessionKey(address target, address from, address to, uint256 amount)
+        external
+        returns (bytes memory returnData)
+    {
+        bytes memory data = abi.encodeWithSelector(TRANSFERFROM_SELECTOR, from, to, amount);
+        returnData = IPluginExecutor(msg.sender).executeFromPluginExternal(target, 0, data);
+    }
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Plugin interface functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc BasePlugin
+    function onInstall(bytes calldata data) external override {
+        if (data.length != 0) {
+            // Call to removeTemporaryOwnerBatch function in BaseSessionKeyPlugin
+            (bool success, bytes memory returnData) = address(msg.sender).call(
+                abi.encodeWithSelector(IPluginExecutor.executeFromPlugin.selector, data));
+            if (!success) {
+                assembly ("memory-safe") {
+                    revert(add(returnData, 32), mload(returnData))
+                }
+            }
+        }
+    }  
+
+    /// @inheritdoc BasePlugin
+    function onUninstall(bytes calldata data) external override {
+        if (data.length != 0) {
+            // Call to removeTemporaryOwnerBatch function in BaseSessionKeyPlugin
+            (bool success, bytes memory returnData) = address(msg.sender).call(
+                abi.encodeWithSelector(IPluginExecutor.executeFromPlugin.selector, data));
+            if (!success) {
+                assembly ("memory-safe") {
+                    revert(add(returnData, 32), mload(returnData))
+                }
+            }
+        }
+    }
```

Comment:
@jaypaik I got a question on the use of `executeFromPlugin` at `onUninstall` function. I was thinking of a solution that modifies session keys through calls to `BaseSessionKeyPlugin` by `executeFromPlugin`, in the (un)installatation process of each child plugin (such as TokenSessionKey). The installation process worked  well, but there was a problem at the uninstallation. Since the `permittedExecutionSelectors` are initialized before the invocation of `onUninstall`, the above function reverts with `ExecFromPluginNotPermitted` error.

Could you let me know if there are any additional contexts or security concerns you considered in the design process, about putting `onUninstall` at the last part of `uninstallPlugin`?